src/overlays/actors/ovl_En_Holl/z_en_holl.c

```diff
@@ -24,37 +36,265 @@ const ActorInit En_Holl_InitVars = {
     (ActorFunc)EnHoll_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_8089A5D0[] = {
+static UNK_TYPE D_8089A550[] = { 0x465055F0, 0x00000000, 0x08000800, 0xFFFFFFFF, 0xB9B055F0, 0x00000000,
+                                 0x00000800, 0xFFFFFFFF, 0xB9B0AA10, 0x00000000, 0x00000000, 0xFFFFFFFF,
+                                 0x4650AA10, 0x00000000, 0x08000000, 0xFFFFFFFF };
+
+static UNK_TYPE D_8089A590[] = { 0xD7000000,  0xFFFFFFFF, 0xFCFFFFFF, 0xFFFDF638, 0x01004008,
+                                 &D_8089A550, 0x06000204, 0x00000406, 0xDF000000, 0x00000000 };
+
+static EnHoll* sThis = NULL;
+
+static EnHollActionFunc sActionFuncs[] = { EnHoll_VisibleIdle, EnHoll_VerticalIdle, EnHoll_TransparentIdle,
+                                           EnHoll_VerticalBgCoverIdle, EnHoll_VisibleIdle };
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 400, ICHAIN_STOP),
 };
 
-#endif
+static f32 sActivationPlaneDistance = 200.0f;
+static f32 sLoadingPlaneDistance = 150.0f;
+static f32 sTranslucencyPlaneDistance = 100.0f;
+static f32 sTransparencyPlaneDistance = 50.0f;
+
+void EnHoll_SetupAction(EnHoll* this) {
+    this->type = EN_HOLL_GET_TYPE(this);
+    this->actionFunc = sActionFuncs[this->type];
+    if (EN_HOLL_IS_VISIBLE(this)) {
+        this->alpha = 255;
+    } else {
+        this->actor.draw = NULL;
+    }
+}
+
+void EnHoll_SetPlayerSide(GlobalContext* globalCtx, EnHoll* this, Vec3f* rotatedPlayerPos) {
+    Player* player = PLAYER;
+
+    /* rotatedPlayerPos = function output
+     * = hypothetical rotation of Player around Holl's y-axis to intersect its y-z plane */
+    Actor_CalcOffsetOrientedToDrawRotation(&this->actor, rotatedPlayerPos, &player->actor.world.pos);
+    this->playerSide = (rotatedPlayerPos->z < 0.0f) ? EN_HOLL_BEHIND : EN_HOLL_BEFORE;
+}
+
+void EnHoll_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnHoll* this = THIS;
+    UNK_TYPE pad;
```

Comment:
```suggestion
    s32 pad;
```
For consistency with the rest of the codebase. Applies elsewhere.